tools/generate_cli_documentation.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+"""
+Script to generate CLI documentation from the argument parser.
+Usage:
+    python generate_cli_documentation.py
+"""
+
+import sys
+from pathlib import Path
+
+
+def generate_cli_docs():
+    # Add the src directory to the path so we can import the modules
+    sys.path.insert(0, str(find_project_root() / "src"))
+    from pysonar_scanner.configuration.cli import CliConfigurationLoader
+
+    """Generate markdown documentation for CLI arguments."""
+    parser = CliConfigurationLoader._CliConfigurationLoader__create_parser()
+
+    # Group arguments by category
+    categories = {
+        "Authentication": ["token", "sonar_host_url", "sonar_region"],
+        "Project Configuration": [
+            "sonar_project_key",
+            "sonar_project_name",
+            "sonar_project_version",
+            "sonar_project_description",
+            "sonar_project_base_dir",
+            "sonar_sources",
+            "sonar_tests",
+        ],
+        "Analysis Configuration": ["verbose", "sonar_python_version", "sonar_filesize_limit"],
+        "Report Integration": [
+            "sonar_python_coverage_report_paths",
+            "sonar_python_pylint_report_path",
+            "sonar_sarif_report_paths",
+            "sonar_python_xunit_report_path",
+            "sonar_python_xunit_skip_details",
+            "sonar_python_mypy_report_paths",
+            "sonar_python_bandit_report_paths",
+            "sonar_python_flake8_report_paths",
+            "sonar_python_ruff_report_paths",
+            "sonar_external_issues_report_paths",
+            "coverage_report_paths",
+            "pylint_report_path",
+            "xunit_report_path",
+            "xunit_skip_details",
+            "mypy_report_paths",
+            "bandit_report_paths",
+            "flake8_report_paths",
+            "ruff_report_paths",
+        ],
+        "Other": [],  # Will contain everything else
```

Comment:
I'd like to see a Python specific section with : 
--sonar-cpd-python-minimum-lines
--sonar-cpd-python-minimum-tokens
--sonar-python-skip-unchanged
--toml-path
--sonar-python-skip-unchanged
--sonar-python-xunit-skip-details
--sonar-python-version